### PR TITLE
feat: add ultimate express

### PR DESCRIPTION
**Pull request by Pho3niX90**
addition of ultimate express https://github.com/dimdenGD/ultimate-express

### CODE REVIEW

File: frameworks/JavaScript/ultimate-express/src/server.mjs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+let db;
+const { DATABASE } = process.env;
+if (DATABASE) db = await import(`./database/${DATABASE}.mjs`);
+
+const extra = { id: 0, message: "Additional fortune added at request time." };
+
+const app = express();
+
+app.get("/plaintext", (req, res) => {
+  writeResponse(res, GREETING, headerTypes["plain"]);
```
Comment (dimdenGD) [inline frameworks/JavaScript/ultimate-express/src/server.mjs:24]: If you write it as instead
```js
app.get("/plaintext", (req, res) => {
  res.setHeader("Content-Type", "text/plain");
  res.setHeader("Server", "UltimateExpress");
  res.send("Hello, World!");
});
```
UltimateExpress will be able to compile it into declarative rule and it'll be much faster

File: frameworks/JavaScript/ultimate-express/src/utils.mjs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export const headerTypes = {
+  plain: "text/plain",
+  json: "application/json",
+  html: "text/html; charset=UTF-8",
+};
+
+export function writeResponse(res, text, type = headerTypes["json"]) {
+  res.writeHead(200, {
+    "content-type": type,
+    server: "Express",
```
Comment (dimdenGD) [inline frameworks/JavaScript/ultimate-express/src/utils.mjs:14]: Isn't it supposed to be UltimateExpress?

File: frameworks/JavaScript/ultimate-express/src/server.mjs (hunk 1 above)
Comment (Pho3niX90) [reply frameworks/JavaScript/ultimate-express/src/server.mjs]: Thanks for letting me know. I have made the changes

File: frameworks/JavaScript/ultimate-express/benchmark_config.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
+{
+  "framework": "ultimate-express",
+  "tests": [{
+    "default": {
+      "json_url": "/json",
+      "plaintext_url": "/plaintext",
```
Comment (joanhey) [inline frameworks/JavaScript/ultimate-express/benchmark_config.json:6]: Only json and plaintext test here !!
It' the same with PostgreSQL. !!!

File: frameworks/JavaScript/ultimate-express/benchmark_config.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      "database": "MongoDB",
+      "framework": "ultimate-express",
+      "language": "JavaScript",
+      "flavor": "NodeJS",
+      "platform": "nodejs",
+      "webserver": "None",
+      "orm": "Full",
+      "os": "Linux",
+      "database_os": "Linux",
+      "display_name": "ultimate-express",
```
Comment (joanhey) [inline frameworks/JavaScript/ultimate-express/benchmark_config.json:53]: 3th Ultimate-express.
Please add in the name to display [normal, charka, mongo, ... ]

**Review by joanhey — CHANGES_REQUESTED** [review]

File: frameworks/JavaScript/ultimate-express/benchmark_config.json (hunk 3 above)
Comment (Pho3niX90) [reply frameworks/JavaScript/ultimate-express/benchmark_config.json:6]: I am not fully understanding this comment.

File: frameworks/JavaScript/ultimate-express/mongodb-app.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    return world;
+  };
+
+  // Configuration
+  app.use(bodyParser.urlencoded({extended: true}));
+
+  // Set headers for all routes
+  app.use((req, res, next) => {
+    res.setHeader("Server", "Express");
```
Comment (dimdenGD) [inline frameworks/JavaScript/ultimate-express/mongodb-app.js:60]: Should be UltimateExpress here too

File: frameworks/JavaScript/ultimate-express/routes.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    app.get('/plaintext', (req, res) => {
+    
+        const responseTxt = "Hello, World!";
+    
+        res.setHeader('Content-Length', responseTxt.length);
+        res.setHeader('Server', 'UltimateExpress-GraphQL-MySQL');
+        res.contentType('text/plain');
+        res.status(200);
+    
+        res.send(responseTxt);
```
Comment (dimdenGD) [inline frameworks/JavaScript/ultimate-express/routes.js:124]: If Content-Length is not specifically needed here and Chunked Transfer encoding is fine, then 
```js
 app.get('/plaintext', (req, res) => {
        res.setHeader("Server", "UltimateExpress-GraphQL-MySQL");
        res.setHeader("Content-Type", "text/plain");

        res.send("Hello, World!");
});
```
would compile it into declarative

**Comment by dimdenGD** [comment]
Also add `app.set("etag", false);` in all tests so it doesnt spend most of the time hashing requests

**Comment by Pho3niX90** [comment]
> Also add `app.set("etag", false);` in all tests so it doesnt spend most of the time hashing requests

should be good now, replaced body parser with express. as well

**Comment by dimdenGD** [comment]
Can I get write access to your fork so I can make the changes

File: frameworks/JavaScript/ultimate-express/app.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ import { sjs, attr } from 'slow-json-stringify'
 const { DATABASE } = process.env;
 const db = DATABASE ? await import(`./database/${DATABASE}.js`) : null;
 
-const jsonSerializer = sjs({ message: attr("string")}); 
+const jsonSerializer = sjs({ message: attr("string") });
```
[the remark this answers is not in the document]
Comment (dimdenGD) [reply frameworks/JavaScript/ultimate-express/app.js:10]: I can't find a benchmark for that

[the remark this answers is not in the document]
Comment (nigrosimone) [reply frameworks/JavaScript/ultimate-express/app.js:10]: I have made it here https://stackblitz.com/edit/stackblitz-starters-rlrggo 

```
slow-json-stringify: 13.105ms
fast-json-stringify: string: 0.86ms
fast-json-stringify: unfafe: 0.815ms
JSON.stringify: 0.86ms
```

for small json JSON.stringify is faster, slow-json-stringify is always slow. Side note "fast-json-stringify: unfafe" is raw fast-json-stringify with no special chars support on string

File: frameworks/JavaScript/ultimate-express/app.js [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+      res
+        .header('Content-Type', 'text/html; charset=utf-8')
+        .end(`<!DOCTYPE html><html><head><title>Fortunes</title></head><body><table><tr><th>id</th><th>message</th></tr>${html}</table></body></html>`);
+    } catch (error) {
+      throw error;
+    }
+  })
+
+  let isCachePopulated = false
```
Comment (cesco69) [inline frameworks/JavaScript/ultimate-express/app.js:121]: can be optimized by populating on test init and not on first call, eg:

```js
  let isCachePopulated = false;
  const populate = async() => {
      if (!isCachePopulated) {
        const worlds = await db.getAllWorlds();
        for (let i = 0; i < worlds.length; i++) {
          cache.set(worlds[i].id, worlds[i]);
        }
        isCachePopulated = true;
      }
      return isCachePopulated;
  });
  populate().then(console.log).catch(console.error);

  app.get('/cached-worlds', async (req, res) => {
    res.setHeader('Server', 'UltimateExpress');

    try {
      await populate();
      const count = parseQueries(req.query.count);
      const worlds = new Array(count);

      for (let i = 0; i < count; i++) {
        worlds[i] = cache.get(generateRandomNumber());
      }

      res.json(worlds);
    } catch (error) {
      throw error;
    }
  });
}
```

File: frameworks/JavaScript/ultimate-express/app.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (dimdenGD) [reply frameworks/JavaScript/ultimate-express/app.js]: You must serialize JSON in the request as per test rules. Your benchmark has a bug and doesnt serialize object properly, I fixed and tested it and it seems that fast-json-stringify is still fastest, so I switched it over.

File: frameworks/JavaScript/ultimate-express/app.js (hunk 8 above)
Comment (dimdenGD) [reply frameworks/JavaScript/ultimate-express/app.js:129]: Benchmark has a warm-up period, so it doesn't matter.

**Pull request merged**